src/x509.c

```diff
@@ -5804,26 +5812,22 @@ static int X509PrintExtensions(WOLFSSL_BIO* bio, WOLFSSL_X509* x509, int indent)
  * wolfSSL_X509_print()
  * return WOLFSSL_SUCCESS on success
  */
-static int X509PrintSignature(WOLFSSL_BIO* bio, WOLFSSL_X509* x509, int indent)
+static int X509PrintSignature_ex(WOLFSSL_BIO* bio, byte* sig, int sigSz, int sigNid,
+        int algOnly, int indent)
 {
     char scratch[MAX_WIDTH];
     int scratchLen;
-    unsigned char* sig = NULL;
-    int sigSz = 0;
     WOLFSSL_ASN1_OBJECT* obj = NULL;
     int ret = WOLFSSL_SUCCESS;
     int i;
     char tmp[100];
     int tmpLen = 0;
-    int sigNid;
 
-    wolfSSL_X509_get_signature(x509, NULL, &sigSz);
     if (sigSz <= 0) {
+        XFREE(sig, NULL, DYNAMIC_TYPE_TMP_BUFFER);
```

Comment:
this helper function free'ing the pointer passed in is a potential trap for other engineers working on the code, please either comment the edge case or have the calling function free the memory

Reply:
sig is now free'd in X509PrintSignature()